src/shared/lib/MutationUtils.tsx

```diff
@@ -565,3 +571,66 @@ export function createAnnotatedProteinImpactTypeFilter(
     };
     return filter;
 }
+
+export function createNumericalFilter(getData: (d: Mutation) => number | null) {
+    const filter = (filter: NumericalFilter, mutation: Mutation) => {
+        const value = getData(mutation);
+        const lowerBound = filter.values[0];
+        const upperBound = filter.values[1];
+        return value !== null && value >= lowerBound && value <= upperBound;
+    };
+    return filter;
+}
+
+export function createCategoricalFilter(getData: (d: Mutation) => string) {
+    const filter = (filter: CategoricalFilter, mutation: Mutation) => {
+        const value = getData(mutation);
+        if (filter.values.length < 2 || value === '') {
+            return false;
+        }
+
+        const filterType = filter.values[0] as string;
+        const filterString = filter.values[1] as string;
+        const selections = filter.values[2] as Set<string>;
```

Comment:
This is not ideal and a bit confusing. We should probably come up with a better solution. Ideally `filterType` should be assigned to `filter.type` not to `filter.values[0]`. Filter values should just be the actual values. Filter value of a categorical filter should probably be defined like this (instead of an array of different types) to make things more clear:

```
{
    filterString: string;
    selections: Set<String>;
}
```

This way we can also pass multiple values if we want/need. But with the current design we can only pass 3 values and they actually correspond to a single value.